tutorials/cloud-spanner-aspnet/index.md

```diff
@@ -0,0 +1,107 @@
+
+Add the following methods to your controller.
+
+    . . .
+    [HttpGet]
    public async Task<IActionResult> GetAll()
    {
        var result = new List<Book>();
     
        using (var connection = new SpannerConnection(
            $"Data Source=projects/{_myProject}/instances/myspanner/databases/books"))
        {
            var selectCmd = connection.CreateSelectCommand("SELECT * FROM bookTable");
            using (var reader = await selectCmd.ExecuteReaderAsync())
            {
                while (await reader.ReadAsync())
                {
                    result.Add(new Book
                    {
                        Id = reader.GetFieldValue<string>("ID"),
                        Title = reader.GetFieldValue<string>("Title"),
                        Author = reader.GetFieldValue<string>("Author"),
                        PublishDate = reader.GetFieldValue<DateTime>("PublishDate")
                    });
                }
            }
        }
     
        return Ok(result);
    }
     
    [HttpGet]
    [Route("{id}", Name="GetBookById")]
    public async Task<IActionResult> Get(string id)
    {
        using (var connection = new SpannerConnection(
            $"Data Source=projects/{_myProject}/instances/myspanner/databases/books"))
        {
            var selectCommand =connection.CreateSelectCommand($"SELECT * FROM bookTable WHERE ID='{id}'");
            using (var reader = await selectCommand.ExecuteReaderAsync())
            {
                while (await reader.ReadAsync())
                {
                    return Ok(new Book
                    {
                        Id = reader.GetFieldValue<string>("ID"),
                        Title = reader.GetFieldValue<string>("Title"),
                        Author = reader.GetFieldValue<string>("Author"),
                        PublishDate = reader.GetFieldValue<DateTime>("PublishDate")
                    });
                }
            }
        }
     
        return NotFound();
    }
+    . . .
+
+
+Now, run your program.  Using Postman, perform an HTTP GET on [http://localhost:56777/api/books](http://localhost:56777/api/books) to return all books or [http://localhost:56777/api/books/[ID](http://localhost:56777/api/books/[ID)] to retrieve a single book.
+
+### **[Optional] Deploy to Google App Engine Flex**
+
+You can deploy this Web API project to Google’s App Engine Flex environment and get automatic scaling and management.
+
+Download and install the [Google Cloud Tools for Visual studio extension](https://cloud.google.com/tools/visual-studio/docs/quickstart#install_cloud_tools_for_visual_studio). Then from the main menu, select **Tools**->**Google Cloud** -> **Publish to Google Cloud**.
+
+![image alt text](image_4.png)
+
+Then click **App Engine Flex**, click **Publish** and your website will now be live at [https://YOUR_PROJECT_ID.appspot.com](https://YOUR_PROJECT_ID.appspot.com)!
+
+### **Summary**
```

Comment:
Suggest changing to What's next, as follows: 
## What's next

This basic tutorial does not have everything in a typical ASP.NET application. For enterprise-grade software, you need also to consider transactions, transient faults, and exponential backoff. The Cloud Spanner library for ADO.NET supports handling these issues, and you can learn more in [Getting Started with Cloud Spanner in C#](https://cloud.google.com/spanner/docs/getting-started/csharp/).

The C# Google Cloud Spanner ADO.NET library is in Beta. If you find a bug, please log it on our[ issue tracker](https://github.com/GoogleCloudPlatform/google-cloud-dotnet/issues).